--- a/PLUGINS/src/wirbelscancontrol/Makefile	2011-03-25 16:50:08.000000000 +0100
+++ b/PLUGINS/src/wirbelscancontrol/Makefile	2015-10-29 18:05:38.925085048 +0100
@@ -1,10 +1,11 @@
 #
 # Makefile for a Video Disk Recorder plugin
 #
-# -- v20110324, Winfried Koehler --
+# $Id$
 
 # The official name of this plugin.
 # This name will be used in the '-P...' option of VDR to load the plugin.
+# By default the main source file also carries this name.
 
 PLUGIN = wirbelscancontrol
 
@@ -12,119 +13,141 @@
 
 VERSION = $(shell grep 'static const char \*VERSION *=' $(PLUGIN).c | awk '{ print $$6 }' | sed -e 's/[";]//g')
 
-### The C++ compiler and options:
+### The directory environment:
 
-CXX      ?= g++
-CXXFLAGS ?= -g -O3 -Wall -Woverloaded-virtual
+# Use package data if installed...otherwise assume we're under the VDR source directory:
+PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
+LIBDIR ?= $(call PKGCFG,libdir)
+LOCDIR = $(call PKGCFG,locdir)
+PLGCFG = $(call PKGCFG,plgcfg)
+#
+TMPDIR ?= /tmp
 
-### The directory environment:
+### The compiler options:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
-TMPDIR = /tmp
-
-### Make sure that necessary options are included:
-
-MAKEGLOBAL := $(wildcard $(VDRDIR)/Make.global)
-
-ifeq ($(MAKEGLOBAL),$(VDRDIR)/Make.global)
-
-### vdr version >= 1.7.13:
-# Make.global is required
-# Make.config is optional
-###
-include $(VDRDIR)/Make.global
--include $(VDRDIR)/Make.config
+export CFLAGS   = $(call PKGCFG,cflags)
+export CXXFLAGS = $(call PKGCFG,cxxflags)
 
-else
+### The version number of VDR's plugin API:
+
+APIVERSION = $(call PKGCFG,apiversion)
+ifeq ($(strip $(APIVERSION)),)
+APIVERSION = $(shell grep 'define APIVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+NOCONFIG := 1
+endif
 
-### vdr version < 1.7.13:
-# enshure that at least -fPIC is given
-# Make.config is optional
-###
--include $(VDRDIR)/Make.config
-CFLAGS   += -fPIC
+# backwards compatibility version < 1.7.34
+API1733 := $(shell if [ "$(APIVERSION)" \< "1.7.34" ]; then echo true; fi; )
+
+ifdef API1733
+
+VDRSRC = $(VDRDIR)
+VDRSRC ?= ../../..
+ifeq ($(strip $(LIBDIR)),)
+LIBDIR = $(VDRSRC)/PLUGINS/lib
+endif
+
+ifndef NOCONFIG
+CXXFLAGS = $(call PKGCFG,cflags)
 CXXFLAGS += -fPIC
-DEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+else
+-include $(VDRSRC)/Make.global
+-include $(VDRSRC)/Make.config
+endif
 
-endif 
+export CXXFLAGS
+else
 
-### The version number of VDR's plugin API (taken from VDR's "config.h"):
+### Allow user defined options to overwrite defaults:
 
-APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
+-include $(PLGCFG)
+endif
 
 ### The name of the distribution archive:
 
 ARCHIVE = $(PLUGIN)-$(VERSION)
 PACKAGE = vdr-$(ARCHIVE)
 
-### Includes and Defines (add further entries here):
+### The name of the shared object file:
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+SOFILE = libvdr-$(PLUGIN).so
 
-DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
+### Includes and Defines (add further entries here):
+
+ifdef API1733
+INCLUDES += -I$(VDRSRC)/include
+endif
+
+DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -DVNSI_SERVER_VERSION='"$(VERSION)"'
+
+ifeq ($(DEBUG),1)
+DEFINES += -DDEBUG
+endif
 
 ### The object files (add further files here):
 
 OBJS = $(PLUGIN).o scanmenu.o
 
-### Which Files to uncrustify (add them here)
-UNCRUSTIFY_FILES = $(PLUGIN).c scanmenu.c scanmenu.h
-
 ### The main target:
 
-all: libvdr-$(PLUGIN).so i18n
+ifdef API1733
+all: install-lib
+else
+all: $(SOFILE)
+endif
+#
 
 ### Implicit rules:
 
 %.o: %.c
-	$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) $<
+	$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) -o $@ $<
 
 ### Dependencies:
 
 MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
-	@$(MAKEDEP) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
+	@$(MAKEDEP) $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
 
 -include $(DEPFILE)
 
 ### Internationalization (I18N):
 
 PODIR     = po
-LOCALEDIR = $(VDRDIR)/locale
 I18Npo    = $(wildcard $(PODIR)/*.po)
-I18Nmsgs  = $(addprefix $(LOCALEDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
+I18Nmo    = $(addsuffix .mo, $(foreach file, $(I18Npo), $(basename $(file))))
+I18Nmsgs  = $(addprefix $(DESTDIR)$(LOCDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
 I18Npot   = $(PODIR)/$(PLUGIN).pot
 
 %.mo: %.po
-	@msgfmt -c -o $@ $<
+	msgfmt -c -o $@ $<
 
 $(I18Npot): $(wildcard *.c)
-	@xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --package-name=vdr-$(PLUGIN) --package-version=$(VERSION) --msgid-bugs-address='<see README>' -o $@ $^
+	xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --package-name=vdr-$(PLUGIN) --package-version=$(VERSION) --msgid-bugs-address='<see README>' -o $@ `ls $^`
 
 %.po: $(I18Npot)
-	@msgmerge -U --no-fuzzy-matching --no-wrap --no-location --backup=none -q $@ $<
+	msgmerge -U --no-wrap --no-location --backup=none -q -N $@ $<
 	@touch $@
 
-$(I18Nmsgs): $(LOCALEDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
-	@mkdir -p $(dir $@)
-	@cp $< $@
+$(I18Nmsgs): $(DESTDIR)$(LOCDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
+	install -D -m644 $< $@
 
 .PHONY: i18n
-i18n: $(I18Nmsgs) $(I18Npot)
+i18n: $(I18Nmo) $(I18Npot)
+
+install-i18n: $(I18Nmsgs)
 
 ### Targets:
 
-libvdr-$(PLUGIN).so: $(OBJS)
+$(SOFILE): $(OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) -o $@
-	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
-perm:
-	@chmod 644 *.{c,h} COPYING HISTORY README Makefile po/*
-	@chmod 755 po
+install-lib: $(SOFILE)
+	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+
+install: install-lib install-i18n
 
-dist: $(I18Npo) clean perm
+dist: $(I18Npo) clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
 	@mkdir $(TMPDIR)/$(ARCHIVE)
 	@cp -a * $(TMPDIR)/$(ARCHIVE)
@@ -133,7 +156,7 @@
 	@echo Distribution package created as $(PACKAGE).tgz
 
 clean:
-	@-rm -f $(OBJS) $(DEPFILE) *.so *.tgz core* *~ $(PODIR)/*.mo $(PODIR)/*.pot
+	@-rm -f $(PODIR)/*.mo $(PODIR)/*.pot
+	@-rm -f $(OBJS) $(DEPFILE) *.so *.tgz core* *~
 
-clean_code: clean
-	uncrustify -c ../../../../clean_code/config --replace --no-backup $(UNCRUSTIFY_FILES)
+compile: $(SOFILE)
